packages/fx-core/src/common/spec-parser/utils.ts

```diff
@@ -43,28 +43,29 @@ export function checkParameters(paramObject: OpenAPIV3.ParameterObject[]): Check
 
   for (let i = 0; i < paramObject.length; i++) {
     const param = paramObject[i];
+    const schema = param.schema as OpenAPIV3.SchemaObject;
+
     if (param.in === "header" || param.in === "cookie") {
-      if (param.required) {
+      if (param.required && schema.default === undefined) {
         paramResult.isValid = false;
       }
       continue;
     }
 
-    const schema = param.schema as OpenAPIV3.SchemaObject;
     if (
       schema.type !== "boolean" &&
       schema.type !== "string" &&
       schema.type !== "number" &&
       schema.type !== "integer"
     ) {
-      if (param.required) {
+      if (param.required && schema.default === undefined) {
```

Comment:
can we have a utility method to extract this logic?

Reply:
sure